src/traffic_cop/traffic_cop.cpp

```diff
@@ -313,6 +313,9 @@ std::variant<std::unique_ptr<parser::ParseResult>, common::ErrorData> TrafficCop
                             common::ErrorCode::ERRCODE_SYNTAX_ERROR);
     error.AddField(common::ErrorField::POSITION, std::to_string(e.GetCursorPos()));
     result.emplace<common::ErrorData>(std::move(error));
+  } catch (const BinderException &e) {
```

Comment:
I don't love that it's a BinderException because we're at the parsing stage, but I see why you did it since you want a different error code and you don't have the cursor position. Is there any other approach you can think of?

Reply:
I can't really think of anything else. TBH i did this bug to learn about the parser and binder and things but I am totally not opposed to anything else. Do you think I should put a check that it is that exact binder exception?